.buildkite/docker-compose.yml

```diff
@@ -47,8 +48,15 @@ services:
       args:
         PYTHON_VERSION: "3.7"
 
+  # a service that does nothing to use as a default depends_on for the runners
```

Comment:
Given that we only actually use the 3.6 runner to test neo4j, Thoughts on just defining separate `runner-3.6` vs `runner-3.6-neo4j` services to do this `depends_on` switch instead if that's possible?

Reply:
Unfortunately that doesn't work, because we want to be reusing the cached docker image that we computed in the first step:

https://github.com/stellargraph/stellargraph/blob/c0c25fb27b0572bf9982e71fa696c6f138c8cd43/.buildkite/pipeline.yml#L18-L35

If there's a separate service, the docker-compose plugin won't recognise that the image from `runner-3.6` is appropriate to be used for `runner-3.6-neo4j`. We could potentially do something with an explicit `image: ...` that is automatically filled out with the appropriate value, but that's potentially worse than this hack?

Reply:
makes sense, yeah seems fine to leave it as is 👍